src/plugins/intel_cpu/src/transformations/cpu_opset/common/pass/permute_slice_n_interpolation.cpp

```diff
@@ -0,0 +1,88 @@
+// Copyright (C) 2018-2024 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "permute_slice_n_interpolation.hpp"
+
+#include "openvino/core/rt_info.hpp"
+#include "openvino/op/interpolate.hpp"
+#include "openvino/op/slice.hpp"
+#include "openvino/op/transpose.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+
+#include "itt.hpp"
+#include "utils/general_utils.h"
+
+
+namespace ov {
+
+intel_cpu::PermuteSliceAndInterpolation::PermuteSliceAndInterpolation() {
+    MATCHER_SCOPE(PermuteSliceAndInterpolation);
+    auto interpolate_pattern = pass::pattern::wrap_type<op::v0::Interpolate,
+                                                        op::v4::Interpolate,
+                                                        op::v11::Interpolate>(pass::pattern::has_static_rank());
+
+    matcher_pass_callback callback = [=](pass::pattern::Matcher& m) {
+        const auto& pattern_map = m.get_pattern_value_map();
+
+        auto interpolate = pattern_map.at(interpolate_pattern).get_node_shared_ptr();
+        if (transformation_callback(interpolate)) {
+            return false;
+        }
+
+        if (interpolate->get_output_target_inputs(0).size() != 1) {
+            return false;
+        }
+
+        for (size_t i = 1lu; i < interpolate->get_input_size(); ++i) {
+            if (!is_type<op::v0::Constant>(interpolate->get_input_node_shared_ptr(i))) {
+                return false;
+            }
+        }
+
+        auto in_0 = interpolate->get_input_node_ptr(0);
+
+        if (is_type<op::v1::Transpose>(in_0) && is_type<op::v8::Slice>(in_0->get_input_node_ptr(0))) {
+            auto slice = as_type_ptr<op::v8::Slice>(in_0->get_input_node_shared_ptr(0));
+            if (slice->get_input_size() < 5 ||
+                slice->get_output_size() != 1 ||
+                slice->get_output_target_inputs(0).size() != 1) {
+                return false;
+            }
+            for (size_t i = 1lu; i < slice->get_input_size(); ++i) {
+                if (!is_type<op::v0::Constant>(slice->get_input_node_ptr(i))) {
+                    return false;
+                }
+            }
+            auto in_rank = slice->get_input_partial_shape(0).rank().get_length();
```

Comment:
This should help to fix a build error
```suggestion
            auto in_rank = slice->get_input_partial_shape(0).size();
```

Reply:
Done